C/primitive/elements/env.c

```diff
@@ -515,7 +524,23 @@ extern transaction* elements_simplicity_mallocTransaction(const rawTransaction*
       sha256_hash(&ctx_outputRangeProofsHash, &output[i].rangeProofHash);
       sha256_hash(&ctx_outputSurjectionProofsHash, &output[i].surjectionProofHash);
     }
-    simplicity_assert(numFees == ix_fee);
+
+    simplicity_assert(ix_fee == numFees);
+    rsort(perm, numFees);
+
+    /* Initialize the feeOuputs array from the perm array.
+     * Because the perm array entries are the same size as the feeOutputs array entries, it is safe to initialize one by one.
+     *
+     * In practical C implementations, the feeOutputs array entires are initalized to the same value as the perm array entries.
+     * In practical C implementations, this is a no-op, and generally compiliers are able to see this fact and eliminate this loop.
+     *
+     * We keep the loop in the code just to be pedantic.
+     */
+    for (size_t i = 0; i < numFees; ++i) {
+      static_assert(0 == offsetof(sigOutput, asset.data), "asset ID is not first field of sigOutput.");
+      feeOutputs[i] = (sigOutput*)(uintptr_t)(perm[i]);
```

Comment:
Why cast to `(uintptr_t)` first?

Reply:
This suppresses warning about both casting away `const` and the warning about the change in pointer alignment.

I'll insert a comment about this.